### ISSUE TITLE
False-positive message "Avoid undefined variables" on static prop

### ISSUE THREAD

**Issue by dereuromark** [phpmd/phpmd#734]
- PHPMD version: 2.8.1
- PHP Version: any

### Current Behavior

```php
    protected static $cmsBlockBuffer = [];

        // inside a method
        if (isset(static::$cmsBlockBuffer[$cmsBlockKey])) {
            return static::$cmsBlockBuffer[$cmsBlockKey];
        }
```
Lot of false positive error noise

> Avoid using undefined variables such as '$cmsBlockBuffer' which will lead to PHP notices.

### Expected Behavior
No errors.

**Comment by kylekatarnls** [comment]
Thanks, we're tracking this bug in #714.
